LeakSanitizer: detected memory leaks test_uncompressed_cache_plus_zstd_codec

```
E           Exception: Sanitizer assert found for instance =================================================================
E           ==1==ERROR: LeakSanitizer: detected memory leaks
E           
E           Direct leak of 88 byte(s) in 1 object(s) allocated from:
```

```
test_non_default_compression/test.py::test_uncompressed_cache_plus_zstd_codec
```
https://s3.amazonaws.com/clickhouse-test-reports/0/4b02d83999808ab06320ba82a4530bb958e6b51f/integration_tests__asan__[1_6].html

--- a/src/Compression/CompressionCodecDeflateQpl.cpp
+++ b/src/Compression/CompressionCodecDeflateQpl.cpp
@@ -9,6 +9,7 @@
 #include <Common/logger_useful.h>
 #include "libaccel_config.h"
 #include <Common/MemorySanitizer.h>
+#include <base/scope_guard.h>
 
 namespace DB
 {
@@ -34,6 +35,7 @@ DeflateQplJobHWPool::DeflateQplJobHWPool()
     // loop all configured workqueue size to get maximum job number.
     accfg_ctx * ctx_ptr = nullptr;
     auto ctx_status = accfg_new(&ctx_ptr);
+    SCOPE_EXIT({ accfg_unref(ctx_ptr); });
     if (ctx_status == 0)
     {
         auto * dev_ptr = accfg_device_get_first(ctx_ptr);

